piq/haarpsi.py

```diff
@@ -0,0 +1,210 @@
+            c: float = 30.0, alpha: float = 4.2) -> torch.Tensor:
+    r"""Compute Haar Wavelet-Based Perceptual Similarity
+    Input can by greyscale tensor of colour image with RGB channels order.
+    Args:
+        x: Tensor of shape :math:`(N, C, H, W)` holding an distorted image.
+        y: Tensor of shape :math:`(N, C, H, W)` holding an target image
+        reduction: Specifies the reduction to apply to the output:
+            ``'none'`` | ``'mean'`` | ``'sum'``. ``'none'``: no reduction will be applied,
+            ``'mean'``: the sum of the output will be divided by the number of
+            elements in the output, ``'sum'``: the output will be summed.
+        data_range: The difference between the maximum and minimum of the pixel value,
+            i.e., if for image x it holds min(x) = 0 and max(x) = 1, then data_range = 1.
+            The pixel value interval of both input and output should remain the same.
+        scales: Number of Haar wavelets used for image decomposition.
+        subsample: Flag to apply average pooling before HaarPSI computation. See [1] for details.
+        c: Constant from the paper. See [1] for details
+        alpha: Exponent used for similarity maps weightning. See [1] for details
+
+    Returns:
+        HaarPSI : Wavelet-Based Perceptual Similarity between two tensors
+    
+    References:
+        [1] R. Reisenhofer, S. Bosse, G. Kutyniok & T. Wiegand (2017)
+            'A Haar Wavelet-Based Perceptual Similarity Index for Image Quality Assessment'
+            http://www.math.uni-bremen.de/cda/HaarPSI/publications/HaarPSI_preprint_v4.pdf
+        [2] Code from authors on MATLAB and Python
+            https://github.com/rgcda/haarpsi
+    """
+
+    _validate_input(input_tensors=(x, y), allow_5d=False, scale_weights=None)
+    x, y = _adjust_dimensions(input_tensors=(x, y))
+
+    # Assert minimal image size
+    kernel_size = 2 ** (scales + 1)
+    if x.size(-1) < kernel_size or x.size(-2) < kernel_size:
+        raise ValueError(f'Kernel size can\'t be greater than actual input size. Input size: {x.size()}. '
+                         f'Kernel size: {kernel_size}')
+
+    # Scale images to [0, 255] range as in the paper
+    x = x * 255.0 / float(data_range)
+    y = y * 255.0 / float(data_range)
+
+    # Downscale input to simulates the typical distance between an image and its viewer.
+    if subsample:
+        up_pad = 0
+        down_pad = max(x.shape[2] % 2, x.shape[3] % 2)
+        pad_to_use = [up_pad, down_pad, up_pad, down_pad]
+        x = F.pad(x, pad=pad_to_use)
+        y = F.pad(y, pad=pad_to_use)
+
+        x = F.avg_pool2d(x, kernel_size=2, stride=2, padding=0)
+        y = F.avg_pool2d(y, kernel_size=2, stride=2, padding=0)
+
+    num_channels = x.size(1)
+    # Convert RGB to YIQ color space https://en.wikipedia.org/wiki/YIQ
+    if num_channels == 3:
+        x = rgb2yiq(x)
+        y = rgb2yiq(y)
```

Comment:
The images should be translated to YIQ and then subsampled according to original implementation.